broker/lib/bosh/BoshDirectorClient.js

```diff
@@ -621,24 +621,37 @@ class BoshDirectorClient extends HttpClient {
 
   /*  Task operations */
 
-  getTasks(options) {
+  getTasks(options, fetchDirectorForDeployment) {
     const query = _.assign({
       limit: 1000
     }, options);
+    let shouldFetch = fetchDirectorForDeployment === true;
+    var configPromise = new Promise((resolve, reject) => {
```

Comment:
I think, not required to create a promise. You can directly write like

return Promise
     .try(()=>shouldFetch ? this.getDirectorConfig(options.deployment) : this.primaryConfigs)
      .map(directorConfig => {
+        return this
+          .makeRequestWithConfig({
+            method: 'GET',
+            url: '/tasks',
+            qs: _.pick(query, ['limit', 'state', 'deployment'])
+          }, 200, directorConfig)
+          .then(res => JSON.parse(res.body))
+          .map(task => {
+            task.id = `${options.deployment}_${task.id}`;
+            return task;
+          });
+      })
       .reduce((all_tasks, tasks) => all_tasks.concat(tasks), []);